concent_api/middleman/tests/test_middleman.py

```diff
@@ -185,3 +173,81 @@ def test_that_broken_connection_from_signing_service_is_reported_to_sentry(self)
             fake_client.close()
             self.crash_logger_mock.error.assert_called_once()
             assert_that(self.crash_logger_mock.error.mock_calls[0][1][0]).contains(error_message)
+
+    def test_that_when_signing_service_connection_is_active_subsequent_attempts_will_fail(self, event_loop):
+        self.middleman._is_signing_service_connection_active = True
+        mocked_reader = prepare_mocked_reader(b"some bytes")
+        mocked_writer = prepare_mocked_writer()
+
+        event_loop.run_until_complete(
+            self.middleman._handle_service_connection(mocked_reader, mocked_writer)
+        )
+
+        mocked_writer.close.assert_called_once_with()
+
+    def test_that_when_authentication_is_unsuccessful_then_connection_ends(self, event_loop):
+        self.middleman._is_signing_service_connection_active = False
+        mocked_reader = prepare_mocked_reader(b"some bytes")
+        mocked_writer = prepare_mocked_writer()
+
+        with mock.patch.object(
+            self.middleman,
+            "_authenticate_signing_servcie",
+            new=async_stream_actor_mock(return_value = False)
+        ):
+            event_loop.run_until_complete(
+                self.middleman._handle_service_connection(mocked_reader, mocked_writer)
+            )
+
+            mocked_writer.close.assert_called_once_with()
+
+    def test_that_when_authentication_is_successful_connection_lasts_until_its_end(self, event_loop):
+        with override_settings(
+            CONCENT_PRIVATE_KEY=CONCENT_PRIVATE_KEY,
+            CONCENT_PUBLIC_KEY=CONCENT_PUBLIC_KEY,
+            SIGNING_SERVICE_PUBLIC_KEY=SIGNING_SERVICE_PUBLIC_KEY,
+        ):
+            self.middleman._is_signing_service_connection_active = False
+            some_bytes = b"some bytes"
+            mocked_reader = prepare_mocked_reader(some_bytes, side_effect=asyncio.IncompleteReadError(some_bytes, 123))
```

Comment:
It might be more readable when we would break this function to separate lines

Reply:
done